index.js

```diff
@@ -68,6 +68,45 @@ module.exports = function(version, _options) {
 
             return config.join('');
         },
+        getWayName: function(language, step, options) {
+            var classes = options ? options.classes || [] : [];
+            if (typeof step !== 'object') throw new Error('step must be an Object');
+            if (!Array.isArray(classes)) throw new Error('classes must be an Array or undefined');
+
+            var wayName;
+            var name = step.name || '';
+            var ref = (step.ref || '').split(';')[0];
+
+            // Remove hacks from Mapbox Directions mixing ref into name
+            if (name === step.ref) {
+                // if both are the same we assume that there used to be an empty name, with the ref being filled in for it
+                // we only need to retain the ref then
+                name = '';
+            }
+            name = name.replace(' (' + step.ref + ')', '');
```

Comment:
Maybe it's better to add also `else` here - why it's necessary to search/replace in empty string?

Reply:
If your thinking about perf here @yuryleb, searching an empty string should be really fast. Maybe almost as fast as the if statement check.

I try to have fewer if statements than more as a coding style (though I just copied this code from a few lines down). For this case in particular, I think avoiding the cost of String.search via an if statement is programing from the interpreter rather than other's reading the code.